script/update_payments.pl

```diff
@@ -127,7 +127,7 @@ sub import_payments {
     foreach my $trn (@$transactions) {
         next if(fiddle_payment($trn));
         next if($trn->{trnamt} <= 0);
-        next if($trn->{name} !~ /SM\s?(\d+)/i);
+        next if($trn->{name} !~ /SM\s?[oO]?(\d+)/i);
 
```

Comment:
In fact, instead of handling this here, correct it in `fiddle_payment`.
```
if ($trans->{name} =~ m/SM\s*([oO\d]+)/) {
  my $id = $1;
  my $orig_id = $1;
  $id =~ s/[oO]/0/g;
  $trans->{name} =~ s/SM\s*$orig_id/SM$id/;
}
```

There are almost certainly more elegant ways to write that.  You can also remove the "JAMES C" case as it is now redundant.